tests/E2E Tests/WebAppUiTests/UiTestHelpers.cs

```diff
@@ -324,7 +324,6 @@ internal static bool StartAndVerifyProcessesAreRunning(List<ProcessStartOptions>
                                                 processDataEntry.EnvironmentVariables);
 
                 processes.Add(processDataEntry.ExecutableName, process);
-                Thread.Sleep(5000);

```

Comment:
Is there a way to block the thread on waiting for all processes to start or error? I assume that's the intent behind this sleep.

Reply:
If you can check that all processes have started, you can poll on that condition. Like here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Common/TestCommon.cs#L258

Reply:
Interesting, it looks like something similar is being done with these tests, after the processes are checked to be running https://github.com/AzureAD/microsoft-identity-web/blob/011bd155ce3ddedeb43daa7348e09584cfa20552/tests/E2E%20Tests/WebAppUiTests/WebAppCallsApiCallsGraphLocally.cs#L335